File: src/main/java/org/ivcode/beeboop/plugin/chat/ChatMessage.java
package org.ivcode.beeboop.plugin.chat;

public record ChatMessage (
    Long id,
    Long timestamp,
    Role role,
    String response,
    String thinking,
    String toolName
){
    enum Role {
        SYSTEM,
        USER,
        ASSISTANT,
        TOOL,
        ERROR
    }
}

File: ollama/src/main/kotlin/history/ChatMessage.kt
package org.ivcode.ai.ollama.history

import io.github.ollama4j.models.chat.OllamaChatMessageRole
import io.github.ollama4j.models.chat.OllamaChatToolCalls

data class ChatMessage(
    val role: OllamaChatMessageRole,
    val response: String? = null,
    val thinking: String? = null,
    var toolCalls: MutableList<OllamaChatToolCalls>? = null,
    var timestamp: Long? = null
)